include/wx/uilocale.h

```diff
@@ -138,11 +138,68 @@ class WXDLLIMPEXP_BASE wxUILocale
     // its dtor is not virtual.
     ~wxUILocale();
 
+    // Try to get user's (or OS's) preferred language setting.
+    // Return wxLANGUAGE_UNKNOWN if language-guessing algorithm failed
+    static int GetSystemLanguage();
+
+    // get the encoding used by default for text on this system, returns
+    // wxFONTENCODING_SYSTEM if it couldn't be determined
+    static wxFontEncoding GetSystemEncoding();
+
+    // get the string describing the system encoding, return empty string if
+    // couldn't be determined
+    static wxString GetSystemEncodingName();
+
+    // Retrieve the language info struct for the given language
+    //
+    // Returns NULL if no info found, pointer must *not* be deleted by caller
+    static const wxLanguageInfo* GetLanguageInfo(int lang);
+
+    // Returns language name in English or empty string if the language
+    // is not in database
+    static wxString GetLanguageName(int lang);
+
+    // Returns ISO code ("canonical name") of language or empty string if the
+    // language is not in database
+    static wxString GetLanguageCanonicalName(int lang);
+
+    // Find the language for the given locale string which may be either a
+    // canonical ISO 2 letter language code ("xx"), a language code followed by
+    // the country code ("xx_XX") or a Windows full language name ("Xxxxx...")
+    //
+    // Returns NULL if no info found, pointer must *not* be deleted by caller
+    static const wxLanguageInfo* FindLanguageInfo(const wxString& locale);
+
+    // Add custom language to the list of known languages.
+    // Notes: 1) wxLanguageInfo contains platform-specific data
+    //        2) must be called before Init to have effect
+    static void AddLanguage(const wxLanguageInfo& info);
+
+    // These two methods are for internal use only. First one creates
+    // ms_languagesDB if it doesn't already exist, second one destroys
+    // it.
+    static void CreateLanguagesDB();
+    static void DestroyLanguagesDB();
+
+    // These two methods are for internal use only.
+    // wxLocaleIdent expects script identifiers as listed in ISO 15924.
+    // However, directory names for translation catalogs follow the
+    // Unix convention, using script aliases as listed  in ISO 15924.
+    // First one converts a script name to its alias, second converts
+    // a script alias to its corresponding script name.
+    // Both methods return empty strings, if the script name or alias
+    // couldn't be found.
+    static wxString GetScriptAliasFromName(const wxString& scriptName);
+    static wxString GetScriptNameFromAlias(const wxString& scriptAlias);
+
 private:
     // This ctor is private and exists only for implementation reasons.
     // It takes ownership of the provided pointer.
     explicit wxUILocale(wxUILocaleImpl* impl = NULL) : m_impl(impl) { }
 
+    // copy default table of languages from global static array to
+    // m_languagesInfo, called by InitLanguagesDB
```

Comment:
I think the comment needs to be updated as it says that this function is called by itself.

Reply:
Actually, I simply copied the existing comment without further thoughts. In fact, the comment was wrong even for the previous implementation. I will try to find a better wording.